cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -270,9 +270,9 @@ namespace s3 {
             };
         }
 
-        template<class Visitor, class KeyBucketizer, class PrefixHandler>
-        void do_iterate_type_impl(KeyType key_type,
-                                  Visitor &&visitor,
+        template<class KeyBucketizer, class PrefixHandler>
+        bool do_iterate_type_impl(KeyType key_type,
```

Comment:
Since we're thinking about code formatting, in the case where we have multiple arguments I would prefer the first argument to start on the next line with a single indent, otherwise the indent depends on the length of the function name and files as a whole end up looking a mess

Reply:
Sorry to be boring about it, but my preference is also to have a double indent on parameter lists, which helps disambiguate them from the following code, without needing floating braces

Reply:
I've pushed a change with your preferred formatting for this one, but really it's a mess even within this file let alone across the project. Motivates the automated formatting checks, I suppose.